dagutils/utils.go

```diff
@@ -79,65 +81,85 @@ func addLink(ctx context.Context, ds ipld.DAGService, root *dag.ProtoNode, child
 		return nil, err
 	}
 
-	if err := ds.Add(ctx, root); err != nil {
+	nd, err := root.Finalize()
+	if err != nil {
 		return nil, err
 	}
-	return root, nil
+	if err := ds.Add(ctx, nd); err != nil {
+		return nil, err
+	}
+	return nd, nil
 }
 
 // InsertNodeAtPath inserts a new node in the tree and replaces the current root with the new one.
-func (e *Editor) InsertNodeAtPath(ctx context.Context, pth string, toinsert ipld.Node, create func() *dag.ProtoNode) error {
+func (e *Editor) InsertNodeAtPath(ctx context.Context, pth string, toinsert ipld.Node, create func() *dag.MutableProtoNode) error {
```

Comment:
this seems like a good candidate for the sort of thing where we may want a different interface that plays nicer with the prime view of the world.

Reply:
yea, I'm talking to @warpfork about this. I beleive we want to use the FocusedTransform function to implement this, which I had forgot about before. But I think we may need to provide some enhancements withing go-ipld-prime.